client/lib/products-values/index.js

```diff
@@ -42,7 +42,8 @@ function formatProduct( product ) {
 		is_domain_registration: product.is_domain_registration !== undefined
 			? product.is_domain_registration
 			: product.isDomainRegistration,
-		free_trial: product.free_trial || product.freeTrial
+		free_trial: product.free_trial || product.freeTrial,
+		is_email_verified: Boolean( product.is_email_verified )
```

Comment:
Ah, sorry, didn't know it could be `undefined`. While it is a falsy value, I'd prefer if it was always a boolean. So, in what cases can it be undefined? It looks like you have backend handled, so maybe you just need to add this field to `client/state/receipts/assembler.js`? :)